scipy/optimize/tests/test_bracket.py

```diff
@@ -569,74 +590,150 @@ def test_scalar_with_limit_right(self, xl0, xm0, xr0, xmax, args):
+                # Case 8:
+                # Similar to Case 4 above, but tests default values of xl0 and xr0.
+                None,
+                4.5,
+                None,
+                -26.3,
+                None,
+                (-26.3, 0)
+            ),
         )
     )
-    def test_minima_at_boundary_point(self, xl0, xm0, xr0, xmin, xmax, args):
+    def test_minimum_at_boundary_point(self, xl0, xm0, xr0, xmin, xmax, args):
         f = self.init_f()
         kwargs = self.get_kwargs(xr0=xr0, xmin=xmin, xmax=xmax, args=args)
         result = _bracket_minimum(f, xm0, **kwargs)
         assert result.status == -1
         assert args[0] in (result.xl, result.xr)
         assert result.nfev == f.count
 
-    @pytest.mark.parametrize(
-        "xm0",
-        (
-            np.array([[0.55], [0.58]]),
-            np.array([[0.55, 0.56], [0.57, 0.58]]),
-        ),
-    )
-    @pytest.mark.parametrize(
-        "xl0",
-        (
-            0.2,
-            np.array([[0.2], [0.3]]),
-            np.array([[0.2, 0.3],
-                      [0.3, 0.4]])
-        ),
-    )
-    @pytest.mark.parametrize(
-        "xr0",
-        (
-            0.6,
-            np.array([[0.6], [0.8]]),
-        ),
-    )
-    @pytest.mark.parametrize(
-        "xmin",
-        (
-            np.array([[-np.inf], [-1.0]]),
-            np.array([[-np.inf, -1.0], [-2.0, -3.0]]),
-        )
-    )
-    @pytest.mark.parametrize(
-        "xmax",
-        (
-            np.array([[np.inf], [2.0]]),
-            np.array([[np.inf, 2.0], [3.5, 4.5]]),
-        )
-    )
-    @pytest.mark.parametrize(
-        "args",
-        (
-            (0.0, 0.0),
-            (np.array([[0.0], [-0.1]]), np.array([[0.0], [2.5]])),
-        )
-    )
-    def test_vectorized(self, xl0, xm0, xr0, xmin, xmax, args):
+    @pytest.mark.parametrize('shape', [tuple(), (12, ), (3, 4), (3, 2, 2)])
+    def test_vectorization(self, shape):
+        # Test for correct functionality, output shapes, and dtypes for
+        # various input shapes.
+        a = np.linspace(-0.05, 1.05, 12).reshape(shape) if shape else 0.6
+        args = (a, 0.0)
+        maxiter = 10
+
+        @np.vectorize
+        def bracket_minimum_single(xm0, xl0, xr0, xmin, xmax, factor, a):
+            return _bracket_minimum(self.init_f(), xm0, xl0=xl0, xr0=xr0, xmin=xmin,
+                                    xmax=xmax, factor=factor, maxiter=maxiter,
+                                    args=(a, 0.0))
+
         f = self.init_f()
-        kwargs = self.get_kwargs(xl0=xl0, xr0=xr0, xmin=xmin, xmax=xmax, args=args)
-        result = _bracket_minimum(f, xm0, **kwargs)
-        self.assert_valid_bracket(result)
-        result.nfev == f.count
+
+        rng = np.random.default_rng(2348234)
+        xl0 = -rng.random(size=shape)
+        xr0 = rng.random(size=shape)
+        xm0 = xl0 + rng.random(size=shape) * (xr0 - xl0)
+        xmin, xmax = 1e3*xl0, 1e3*xr0
+        if shape:  # make some elements un
```

Comment:
```suggestion
        if shape:  # make some elements infinite
```
?

Reply:
Copy and pasted from the corresponding `bracket_root` without being read.

https://github.com/scipy/scipy/blob/fb4c65fdf89866728a9762b47e1c75c03e0b2777/scipy/optimize/tests/test_bracket.py#L125